src/lib.rs

```diff
@@ -80,7 +80,11 @@ where
 {
     let addrs = SocketAddrs::from(addrs);
     let local = match addrs {
-        SocketAddrs::Local(local) | SocketAddrs::Both((local, _)) => local,
+        SocketAddrs::Local(mut local) | SocketAddrs::Both((mut local, _)) => {
+            // Let the OS choose an unused local port.
+            local.set_port(0);
```

Comment:
If we drop the user provided local port anyways, why ask the user for a port in the first place? Why not take a local [`IpAddr`](https://doc.rust-lang.org/std/net/enum.IpAddr.html) instead of a `SocketAddr`?

Reply:
We could, now. Let me see if I can plumb this in in a backwards-compatible way.